src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs

```diff
@@ -1079,11 +1079,21 @@ private CachedCodeEntry CacheCode(CachedCodeEntryKey key)
             return newcached;
         }
 
+#if MONO
+	// This is here so we can debug this issue: https://github.com/mono/mono/pull/7982,
+	// once that is fixed, we can remove this.
+	
+	bool enableCompiled = Environment.GetEnvironmentVariable ("MONO_REGEX_COMPILED_ENABLE") != null;
```

Comment:
Indentation is off here, corefx uses spaces.

The variable should be `static readonly` so we don't check it on each Regex instance.

Reply:
This makes it worse for MOBILE profiles like monodroid because it just makes the linker work harder and MONO_REGEX_COMPILED_ENABLE cannot be set there anyway. I'd just return false in all cases